### PR TITLE
device: Add validation and debug

### LINKED ISSUE #172: Add device debug log
Add a log call showing device details.

### PR DESCRIPTION
**Pull request by jodh-intel**
Add a log call showing device details and then validate what
parts of the device can be checked.

Fixes #172.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

**Comment by jodh-intel** [comment]
Related: #171.

File: device.go [hunk 1]
```diff
@@ -276,6 +276,28 @@ func addDevices(devices []*pb.Device, spec *pb.Spec) error {
...
+		agentLog.WithFields(logrus.Fields{
+			"device-id":             device.Id,
+			"device-type":           device.Type,
+			"device-vm-path":        device.VmPath,
+			"device-container-path": device.ContainerPath,
+			"device-options":        device.Options,
+		}).Debug()
+
+		if device.Type == "" {
+			return fmt.Errorf("invalid type for device %v", device)
```
Comment (bergwolf) [inline device.go]: return grpcStatus instead?

Comment (jodh-intel) [reply device.go]: Good catch! Branch updated - now with more test goodness :)

**Comment by devimc** [comment]
lgtm

**Pull request merged**